Shutdown raises exception on Eclipse IDE

When Qute LS is stopped on Eclipse IDE, we have an NPE exactly like in XML Language Server (see https://github.com/eclipse/lemminx/issues/1132)

```
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at com.redhat.qute.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 12 more
Caused by: java.lang.NullPointerException
	at com.redhat.qute.ls.QuteLanguageServer.shutdown(QuteLanguageServer.java:163)
	... 17 more

```

See PR in LemMinX https://github.com/eclipse/lemminx/pull/1134 to fix the same problem in Qute LS.

--- a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteLanguageServer.java
+++ b/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteLanguageServer.java
@@ -117,7 +117,7 @@ public class QuteLanguageServer
 	 * turn on/off
 	 *
 	 * (non-Javadoc)
-	 * 
+	 *
 	 * @see org.eclipse.lsp4j.services.LanguageServer#initialized(org.eclipse.lsp4j.
 	 * InitializedParams)
 	 */
@@ -159,7 +159,7 @@ public class QuteLanguageServer
 
 	@Override
 	public CompletableFuture<Object> shutdown() {
-		if (capabilityManager.getClientCapabilities().getExtendedCapabilities().shouldLanguageServerExitOnShutdown()) {
+		if (capabilityManager.getClientCapabilities().shouldLanguageServerExitOnShutdown()) {
 			ScheduledExecutorService delayer = Executors.newScheduledThreadPool(1);
 			delayer.schedule(() -> exit(0), 1, TimeUnit.SECONDS);
 		}
@@ -176,10 +176,12 @@ public class QuteLanguageServer
 		System.exit(exitCode);
 	}
 
+	@Override
 	public TextDocumentService getTextDocumentService() {
 		return this.textDocumentService;
 	}
 
+	@Override
 	public QuteWorkspaceService getWorkspaceService() {
 		return this.workspaceService;
 	}

--- a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/capabilities/ClientCapabilitiesWrapper.java
+++ b/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/capabilities/ClientCapabilitiesWrapper.java
@@ -40,9 +40,9 @@ public class ClientCapabilitiesWrapper {
 
 	/**
 	 * IMPORTANT
-	 * 
+	 *
 	 * This should be up to date with all Server supported capabilities
-	 * 
+	 *
 	 */
 
 	public boolean isCodeActionDynamicRegistered() {
@@ -84,11 +84,11 @@ public class ClientCapabilitiesWrapper {
 	public boolean isDocumentHighlightDynamicRegistered() {
 		return v3Supported && isDynamicRegistrationSupported(getTextDocument().getDocumentHighlight());
 	}
-	
+
 	public boolean isReferencesDynamicRegistrationSupported() {
 		return v3Supported && isDynamicRegistrationSupported(getTextDocument().getReferences());
 	}
-	
+
 	public boolean isLinkedEditingRangeDynamicRegistered() {
 		return v3Supported && isDynamicRegistrationSupported(getTextDocument().getLinkedEditingRange());
 	}
@@ -96,6 +96,19 @@ public class ClientCapabilitiesWrapper {
 	private boolean isDynamicRegistrationSupported(DynamicRegistrationCapabilities capability) {
 		return capability != null && capability.getDynamicRegistration() != null
 				&& capability.getDynamicRegistration().booleanValue();
+	}
+
+	/**
+	 * Returns true if the client should exit on shutdown() request and avoid
+	 * waiting for an exit() request
+	 *
+	 * @return true if the language server should exit on shutdown() request
+	 */
+	public boolean shouldLanguageServerExitOnShutdown() {
+		if (extendedCapabilities == null) {
+			return false;
+		}
+		return extendedCapabilities.shouldLanguageServerExitOnShutdown();
 	}
 
 	public TextDocumentClientCapabilities getTextDocument() {

